setup/setup.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable */
+"use strict";
+
+class Setup {
+	constructor() {
+		this.fs = require("fs");
+		this.path = require("path");
+		this.root = process.cwd();
+		this.wpfolder = this.path.resolve(this.root, "node_modules/webpack/");
+		this.hasYarn = false;
+		this.msg = {
+			setupStart: "Setup start",
+			setupSuccess: "Setup complete",
+			setupNoSymlink: "Setup failed to establish symlink for webpack",
+			setupFail: "Setup failed",
+			setupSkip: "Skip setup"
+		};
+	}
+
+	run() {
+		let p = new Promise((resolve, reject) => {
+			if (!this.checkSymlinkExists()) {
+				console.log(this.msg.setupStart);
+				this.checkYarnInstalledAsync().then(() => {
+					return this.runSetupAsync();
+				}).catch((e) => {
+					console.error(e);
+					reject();
+				}).then((stdout) => {
+					if (!this.checkSymlinkExists()) {
+						console.error(this.msg.setupNoSymlink);
+						reject();
+					} else {
+						console.log(stdout);
+						console.log(this.msg.setupSuccess);
+						resolve();
+					}
+				}).catch((e) => {
+					console.error(e);
+					console.log(this.msg.setupFail);
+					reject(e);
+				});
+			} else {
+				console.log(this.msg.setupSkip);
+				resolve();
+			}
+		});
+
+		p.then(() => {
+			process.exit(0);
+		}).catch((e) => {
+			process.exit(1);
+		});
+		return p;
+	}
+
+	runSetupAsync() {
+		return new Promise((resolve, reject) => {
+			try {		
+				const pm = this.hasYarn ? "yarn" : "npm";
```

Comment:
npm doesn't work, because the lockfile is in yarn format. Only yarn is supported